### ISSUE TITLE
Dapr Runtime panic for malformed/unexpected workflow URLs

### ISSUE THREAD

**Issue by philliphoff** [dapr/dapr#7087]
## In what area(s)?

/area runtime

> /area operator

> /area placement

> /area docs

> /area test-and-release

## What version of Dapr?

1.12.0

## Expected Behavior

No panics occur.

## Actual Behavior

Dapr runtime panics and, in at least one [seemingly-related case](https://github.com/dapr/dotnet-sdk/issues/1171), crashes.

Sample panic:

```bash
2023/10/23 13:36:19 http: panic serving [::1]:63147: runtime error: index out of range [4] with length 3
goroutine 7776 [running]:
net/http.(*conn).serve.func1()
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:1854 +0xb0
panic({0x108ba6e60, 0x140019c21e0})
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/runtime/panic.go:890 +0x258
github.com/dapr/dapr/pkg/diagnostics.(*httpMetrics).convertPathToMetricLabel(0x14001c205f8?, {0x14001dbf535, 0x19})
        /Users/runner/work/dapr/dapr/pkg/diagnostics/http_monitoring.go:288 +0x56c
github.com/dapr/dapr/pkg/diagnostics.(*httpMetrics).HTTPMiddleware.func1({0x1355400d8?, 0x140010e6730}, 0x14001750100)
        /Users/runner/work/dapr/dapr/pkg/diagnostics/http_monitoring.go:224 +0x98
net/http.HandlerFunc.ServeHTTP(0x140010e6730?, {0x1355400d8?, 0x140010e6730?}, 0x10bf34da8?)
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:2122 +0x38
github.com/dapr/dapr/pkg/diagnostics.HTTPTraceMiddleware.func1({0x109047d00?, 0x140018da0e0}, 0x14001750100)
        /Users/runner/work/dapr/dapr/pkg/diagnostics/http_tracing.go:79 +0x1e0
net/http.HandlerFunc.ServeHTTP(0x140005c3708?, {0x109047d00?, 0x140018da0e0?}, 0x2d1cf50?)
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:2122 +0x38
github.com/dapr/dapr/pkg/http.MaxBodySizeMiddleware.func1.1({0x109047d00, 0x140018da0e0}, 0x14001750100)
        /Users/runner/work/dapr/dapr/pkg/http/middlewares.go:33 +0xe0
net/http.HandlerFunc.ServeHTTP(0x14000666f30?, {0x109047d00?, 0x140018da0e0?}, 0x102f350a8?)
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:2122 +0x38
github.com/dapr/dapr/pkg/http.StripSlashesMiddleware.func1({0x109047d00, 0x140018da0e0}, 0x14001750100)
        /Users/runner/work/dapr/dapr/pkg/http/middlewares.go:96 +0x1a0
net/http.HandlerFunc.ServeHTTP(0x14001dbf535?, {0x109047d00?, 0x140018da0e0?}, 0x10da9a1d8?)
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:2122 +0x38
github.com/dapr/dapr/pkg/http.CleanPathMiddleware.func1({0x109047d00, 0x140018da0e0}, 0x14001750100)
        /Users/runner/work/dapr/dapr/pkg/http/middlewares.go:115 +0xf0
net/http.HandlerFunc.ServeHTTP(0x10904a8d8?, {0x109047d00?, 0x140018da0e0?}, 0x10c0dc710?)
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:2122 +0x38
github.com/go-chi/chi/v5.(*Mux).ServeHTTP(0x14001800d80, {0x109047d00, 0x140018da0e0}, 0x14001750000)
        /Users/runner/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.10/mux.go:90 +0x2e4
golang.org/x/net/http2/h2c.h2cHandler.ServeHTTP({{0x109001f20?, 0x14001800d80?}, 0x140018482d0?}, {0x109047d00, 0x140018da0e0}, 0x14001750000)
        /Users/runner/go/pkg/mod/golang.org/x/net@v0.15.0/http2/h2c/h2c.go:125 +0x424
net/http.serverHandler.ServeHTTP({0x109029728?}, {0x109047d00, 0x140018da0e0}, 0x14001750000)
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:2936 +0x2d8
net/http.(*conn).serve(0x140018e8480, {0x10904a980, 0x140005a1470})
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:1995 +0x560
created by net/http.(*Server).Serve
        /Users/runner/hostedtoolcache/go/1.20.8/x64/src/net/http/server.go:3089 +0x520
```

## Steps to Reproduce the Problem

1. Run the workflow [example](https://github.com/dapr/dotnet-sdk/blob/3b979e6bdb1d779563f1656fa684183b2bfecd08/examples/Workflow/README.md) from the Dapr .NET SDK
1. Execute a (malformed) HTTP method invocation on the workflow API, such as `PUT http://localhost:3500/v1.0-beta1/workflows/foo`
1. Note the logged panic (if not crash)

This appears to be coming from an incorrect assumption regarding an array length in the [HTTP monitoring logic for workflows](https://github.com/dapr/dapr/blob/ce846a389ca285e58d9a1142b6fd3f4e35b5f0d8/pkg/diagnostics/http_monitoring.go#L288), which is new to 1.12. 

## Release Note

RELEASE NOTE:

**Comment by JoshVanL** [comment]
@philliphoff can this now be closed?

**Comment by philliphoff** [comment]
Closed with #7094.
